### ISSUE TITLE
Show table of contents in each Rev News post

### ISSUE THREAD

**Issue by tfnico** [git/git.github.io#44]
Since they are quite long..

**Comment by tfnico** [comment]
Will require a plugin it seems: https://github.com/dafi/jekyll-toc-generator

**Comment by dscho** [comment]
You can always use a local script to generate a TOC from an existing Markdown file, e.g. using https://github.com/jonschlinkert/markdown-toc. Of course, this would be a manual step, but publishing Rev News is already a manual step, isn't it? BTW is it documented anywhere what the steps are? Or even better: is there a script already that performs the publication?

**Comment by tfnico** [comment]
@chriscool would have to comment on the publishing process. I suspect it's a manual move of the draft into the right place/name and commit & push, then a mail written to the Git list with some copy/pasted content. Then comes the mailchimp routine later (@mjaix takes care of that these days).

Long story short, adding a toc command in there should be possible, but yeah, we should document the process then.

**Comment by chriscool** [comment]
Yeah, it is a manual process and @tfnico just documented it at a high level. I could automate some trivial steps, but at least at the beginning it will still need significant manual work for example to make sure the draft is sent to all the people who are concerned.

**Comment by dscho** [comment]
On Sat, 18 Nov 2017, Christian Couder wrote:

> Yeah, it is a manual process and @tfnico just documented it at a high level.

That's good, that it is documented at last.

> I could automate some trivial steps,

Please do.

> but at least at the beginning it will still need significant manual work for example to make sure the draft is sent to all the people who are concerned.

Why should that be manual? Would it not make for a much more robust
process if there was an explicit list "Mentioned people:" in the text, and
a script to collected the respective emails, say, from Git's .mailmap (and
possibly a new .mailmap in the git.github.io repository)?

In other words: why not make the process a lot more transparent (and
thereby automatable).

**Comment by tfnico** [comment]
Oh hey, I just found some more process has been written here already:

https://git.github.io/rev_news/news_sources/

I'll patch up that page a bit.

**Comment by allejo** [comment]
Fast forward to 2019, there's now https://github.com/allejo/jekyll-toc that would let you achieve this on GitHub Pages without a plug-in or JS. Is this still something this project would be interested in?

**Comment by chriscool** [comment]
Yeah, I think it would be interesting to have a table of contents. Thanks your suggestion!
